lib/ExportPropertyMainTemplatePlugin.js

```diff
@@ -6,15 +6,28 @@
 
 const { ConcatSource } = require("webpack-sources");
 
+/** @typedef {import("./Compilation")} Compilation */
+
+/**
+ * @param {string[]} accessor the accessor to convert to path
+ * @returns {string} the path
+ */
 const accessorToObjectAccess = accessor => {
 	return accessor.map(a => `[${JSON.stringify(a)}]`).join("");
 };
 
 class ExportPropertyMainTemplatePlugin {
+	/**
+	 * @param {string|string[]} property the name of the property to export
+	 */
 	constructor(property) {
 		this.property = property;
 	}
 
+	/**
+	 * @param {Compilation} compilation the compilation instance
+	 * @returns {void}
+	 */
 	apply(compilation) {
 		const { mainTemplate, chunkTemplate } = compilation;
 
```

Comment:
`onRenderWithEntry` is missing types (I guess you're more focused on public API for now?)